test/kubernetes_provider/kubernetes_provider_test.go

```diff
@@ -44,7 +45,7 @@ func TestKubernetes_Provider(t *testing.T) {
 		kubernetesProvider, ok = provider.(*kubernetessecrets.Provider)
 		So(ok, ShouldBeTrue)
 
-		kubernetesProvider.Client = testClientSet
+		kubernetesProvider.SecretsClient = testSecretsClient
```

Comment:
We shouldn't be putting "setup" code for the remaining _sibling_ tests inside of the first test.  Each test case should run independently.

It looks like the pattern GoConvey uses for shared setup is to place it into a container convey block, one level up from the tests being run: 

https://github.com/smartystreets/goconvey/blob/master/examples/bowling_game_test.go#L20

I still think we should remove Convey as a dep, but that's probably out of scope for this PR.